Makefile

```diff
@@ -227,5 +240,8 @@ upload-tarballs:
 src/allmydata/_version.py:
 	$(MAKE) make-version
 
-.tox: tox.ini setup.py
-	tox --notest -p all
+.tox/create-venvs.log: tox.ini setup.py
+	tox --notest -p all | tee -a "$(@)"
+
+$(VCS_HOOKS): .tox/create-venvs.log .pre-commit-config.yaml
+	"./$(<)/py36/bin/pre-commit" install --hook-type $(@:.git/hooks/%=%)
```

Comment:
This seems very tied to a particular setup (that is, having a virtualenv in the root named `py36`). It's probably better to invoke it as just `pre-commit` or `python -m pre-commit` (if that works for this package) so that the developer can worry about how/where their environments are defined.

Reply:
That command happened to be broken, I've just pushed a fix.

Regardless, the only set up it's tied to is that same set up this project uses for testing, namely tox and the venvs it creates and maintains.  In that sense it's *less* dependent on developer set up.

Reply:
oh, this somehow points into `./.tox`? (This is beyond my makefile knowledge ;) )

Reply:
Ah!  Yeah, `$(<)` refers to the first "prerequisite" of the target, `.tox/create-venvs.log` in this case.  The (now pushed) fixed version is `$(dir $(<))` which in this case yields `.tox/`.